### ISSUE TITLE
403 Access Denied on Macy's website

### ISSUE THREAD

**Issue by sean-hill** [puppeteer/puppeteer#2511]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.3.0
* Platform / OS version: MacOS
* URLs (if applicable): https://www.macys.com/account/signin
* Node.js version: 8.2.1

**What steps will reproduce the problem?**

```js
const browser = await puppeteer.launch({ headless: false })
const page = await browser.newPage()
await page.goto('https://www.macys.com/account/signin')

await page.type('#email', 'tester@tester.tester')
await page.type('#pw-input', 'testtest')
await page.click('#sign-in')
await page.waitFor(5000)

await browser.close()
```

1. Go to Macy's sign-in page
2. Sign in with my test account
3. Notice the Access Denied page afterwards

**What is the expected result?**

That I would be logged into Macy's account and see my name "Test" in the top right like the below image.

![image](https://user-images.githubusercontent.com/1205866/39729110-fde11110-520e-11e8-9d58-09fea75218e3.png)

**What happens instead?**

I'm presented with a 403 Access Denied page.

![6c0720f3-39a9-4818-8ca9-7d3a6eb0ba96_full](https://user-images.githubusercontent.com/1205866/39729133-1ccbc00c-520f-11e8-8769-53133495ee99.png)

**Comment by sean-hill** [comment]
I've been working on this and notice that if I just open the downloaded chromium binary in my `node_modules` folder and replicate the same steps myself with Chromium, that it works just fine. It's only when it's being automated that I see this issue arise.

**Comment by sean-hill** [comment]
I tried setting the `ignoreDefaultArgs: true` and it worked! So something with the default args causes this page return `Access Denied`, I'll look into which one.

**Comment by sean-hill** [comment]
I noticed that setting `ignoreDefaultArgs: true` allows it to work. Specifically removing the `enable-automation` flag as an argument allows it to work.

**Comment by sean-hill** [comment]
It looks like Macy's is using fancy bot detection. I ended up following [this tutorial](https://intoli.com/blog/not-possible-to-block-chrome-headless/) and it helped solve my issues.

**Comment by mymikemiller** [comment]
Wow, thank you @sean-hill ! I spent 4 days trying to automate logging into smithsfoodanddrug.com until I finally found your solution! Smiths is using the same [Akamai bot detection](https://www.akamai.com/us/en/products/cloud-security/bot-manager.jsp) as Macys.

I sometimes still get an error on login, but that goes away when I also disable the 'headless' flag as you did in your repro steps. It causes the browser to pop up when the script runs, which is unfortunate, but login always succeeds in that case.

Thanks for posting your findings here!
